src/main/java/de/zalando/ep/zalenium/aspect/HubAspect.aj

```diff
@@ -0,0 +1,51 @@
+package de.zalando.ep.zalenium.aspect;
+
+
+import java.util.EnumSet;
+
+import javax.servlet.DispatcherType;
+
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.handler.StatisticsHandler;
+import org.openqa.grid.web.Hub;
+import org.seleniumhq.jetty9.servlet.FilterHolder;
+import org.seleniumhq.jetty9.servlet.ServletContextHandler;
+import org.seleniumhq.jetty9.servlet.ServletHolder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.prometheus.client.exporter.MetricsServlet;
+import io.prometheus.client.filter.MetricsFilter;
+import io.prometheus.client.hotspot.DefaultExports;
+import io.prometheus.client.jetty.JettyStatisticsCollector;
+
+privileged public aspect HubAspect {
+    private static final Logger log = LoggerFactory.getLogger(HubAspect.class.getName());
+
+    pointcut callAddDefaultServlets(ServletContextHandler handler, Hub hub) :
+        call (private void Hub.addDefaultServlets(ServletContextHandler)) && args(handler) && target(hub);
+    
+    before(ServletContextHandler handler, Hub hub) : callAddDefaultServlets(handler, hub) {
+        log.info("Registering custom Zalenium servlets");
+        
+        FilterHolder prometheusMetricsFilter = handler.addFilter(MetricsFilter.class, "/*", EnumSet.allOf(DispatcherType.class));
+        prometheusMetricsFilter.setInitParameter("metric-name", "webapp_metrics_filter");
+        prometheusMetricsFilter.setInitParameter("help", "This is the help for your metrics filter");
+        prometheusMetricsFilter.setInitParameter("buckets", "0.005,0.01,0.025,0.05,0.075,0.1,0.25,0.5,0.75,1,2.5,5,7.5,10");
+        
+        // Enable prometheus metrics at /metrics
+        handler.addServlet(new ServletHolder(new MetricsServlet()), "/metrics");
```

Comment:
What will this do is selenium has a `/metrics`? Can we / should we check that?

Reply:
Have a look in Hub.class in selenium. That’s where it registers all it’s servlets. 
I don’t recall seeing a metrics. Plus mine is registered first so I win ;)

Not at my pc, so can’t check right now.

Reply:
No `/metrics` endpoint:

https://github.com/SeleniumHQ/selenium/blob/157703a475f8061a57deb489b746757e7efc0040/java/server/src/org/openqa/grid/web/Hub.java#L123-L162